### PR TITLE
TF-610: As a user, I want to see list of filter email rules 

### PR DESCRIPTION
**Pull request by ManhNTX**
Need merge https://github.com/linagora/tmail-flutter/pull/808 before review

https://user-images.githubusercontent.com/107173849/183373538-41e0cc6c-4756-465d-a667-457d651afd88.mov

https://user-images.githubusercontent.com/107173849/183373585-55b2113e-e5e8-4e04-8d02-51a23d858489.mov

### CODE REVIEW

File: core/lib/presentation/views/bottom_popup/cupertino_action_sheet_action_builder.dart [hunk 1]
```diff
@@ -6,24 +6,25 @@ import 'package:flutter_svg/flutter_svg.dart';
...
-  CupertinoActionSheetActionBuilder(
-    this.key,
-    this.actionIcon,
-    this.actionName
-  );
+  CupertinoActionSheetActionBuilder(this.key, this.actionIcon, this.actionName);
 
-  void onActionClick(OnCupertinoActionSheetActionClick<T> onCupertinoActionSheetActionClick) {
+  void onActionClick(
+      OnCupertinoActionSheetActionClick<T> onCupertinoActionSheetActionClick) {
```
Comment (hoangdat) [inline core/lib/presentation/views/bottom_popup/cupertino_action_sheet_action_builder.dart:21]: I am not fan of this style

File: lib/features/manage_account/presentation/manage_account_dashboard_view.dart [hunk 2]
```diff
@@ -146,6 +147,8 @@ class ManageAccountDashBoardView extends GetWidget<ManageAccountDashBoardControl
           return ProfilesView();
         case AccountMenuItem.languageAndRegion:
           return LanguageAndRegionView();
+        case AccountMenuItem.emailRules:
+          return EmailRulesView();
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/manage_account_dashboard_view.dart:151]: please check the session to show/or not show this item. It is an extension of Linagora. It means that other JMAP server may not include it.

File: lib/features/manage_account/presentation/profiles/profiles_bindings.dart [hunk 3]
```diff
@@ -9,6 +10,7 @@ class ProfileBindings extends BaseBindings {
   void dependencies() {
     super.dependencies();
     IdentitiesBindings().dependencies();
+    EmailRulesBindings().dependencies();
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/profiles/profiles_bindings.dart:13]: check the extension is available or not to binding its componentss

File: lib/features/manage_account/presentation/email_rules/email_rules_controller.dart [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import 'package:tmail_ui_user/features/manage_account/presentation/manage_account_dashboard_controller.dart';
+import 'package:tmail_ui_user/main/localizations/app_localizations.dart';
+
+class EmailRulesController extends BaseController {
+  final listEmailRule = <TMailRule>[].obs;
+  final _accountDashBoardController =
+      Get.find<ManageAccountDashBoardController>();
+  final GetAllRulesInteractor _getAllRulesInteractor;
+  final _imagePaths = Get.find<ImagePaths>();
+  late Worker accountIdWorker;
```
Comment (dab246) [inline lib/features/manage_account/presentation/email_rules/email_rules_controller.dart:21]: why you not use it?

File: lib/features/manage_account/presentation/email_rules/email_rules_view.dart [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      ),
+    );
+  }
+
+  Widget _buildLoadingView() {
+    return Obx(() => controller.viewState.value.fold(
+            (failure) => const SizedBox.shrink(),
+            (success) => success is LoadingState
+            ? Padding(
+            padding: const EdgeInsets.only(bottom: 24),
```
Comment (dab246) [inline lib/features/manage_account/presentation/email_rules/email_rules_view.dart:60]: should be indented

File: lib/features/manage_account/presentation/email_rules/email_rules_view.dart [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            child: loadingWidget)
```
Comment (dab246) [inline lib/features/manage_account/presentation/email_rules/email_rules_view.dart:61]: same

**Review by dab246 — COMMENTED** [review]
You should check the condition whether or not support `rule_filter` capability in session to inject it

File: lib/features/manage_account/presentation/manage_account_dashboard_controller.dart [hunk 7]
```diff
@@ -127,4 +129,12 @@ class ManageAccountDashBoardController extends ReloadableController {
       pushAndPopAll(AppRoutes.MAILBOX_DASHBOARD, arguments: sessionCurrent);
     }
   }
+
+  bool checkAvailableRuleFilterInSession() {
+    if(sessionCurrent?.capabilities.containsKey(capabilityRuleFilter) == true) {
+      EmailRulesBindings().dependencies();
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/manage_account_dashboard_controller.dart:135]: it is not make-sense. Checking function, but have the dependencies logic?

Comment (hoangdat) [inline lib/features/manage_account/presentation/manage_account_dashboard_controller.dart:135]: why not use `EmailRuleView` as a page, `dependencies` is controlled by `binding class`?

**Review by dab246 — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**